Add legend item visibility to user guide

tcmetzger opened the pull request:
This PR adds a section on toggling visibility for legend items. This is a follow-up to #11412 and also fixes #9261.
cc @bryevdv

bryevdv commented:
It's probably worth mentioning also that if the legend item is a "field" type that groups on the client (browser) side, then *all* generated items that might have been shown will all be suppressed

tcmetzger commented:
@bryevdv I added a note on browser-side grouping. I also updated `_docstrings.py` because I saw that there were still some hard links (to `docs.boke.org` instead of `:ref:`) and some typos. I added a link to the new user guide section on visibility to `annotations.py` as well.
